implementations/rust/ockam/ockam_api/src/cli_state/projects.rs

```diff
@@ -1,56 +1,103 @@
+use ockam::identity::IdentitiesVerification;
 use std::collections::HashMap;
+use std::sync::Arc;
 
 use ockam_core::errcode::{Kind, Origin};
 use ockam_core::Error;
+use ockam_vault::SoftwareVaultForVerifyingSignatures;
 
 use crate::cli_state::CliState;
+use crate::cloud::project::models::ProjectModel;
 use crate::cloud::project::Project;
+use crate::ProjectsRepository;
 
 use super::Result;
 
-impl CliState {
-    #[instrument(skip_all, fields(project_id = project.id))]
-    pub async fn store_project(&self, project: Project) -> Result<()> {
-        let repository = self.projects_repository();
-        repository.store_project(&project).await?;
+pub struct Projects {
+    projects_repository: Arc<dyn ProjectsRepository>,
+    identities_verification: IdentitiesVerification,
+}
+
+impl Projects {
+    pub fn new(
+        projects_repository: Arc<dyn ProjectsRepository>,
+        identities_verification: IdentitiesVerification,
+    ) -> Self {
+        Self {
+            projects_repository,
+            identities_verification,
+        }
+    }
+
+    #[instrument(skip_all, fields(project_id = project_model.id))]
+    pub async fn import_and_store_project(&self, project_model: ProjectModel) -> Result<Project> {
+        let project = Project::import(project_model.clone()).await?;
+        self.store_project(project).await
+    }
+
+    #[instrument(skip_all, fields(project_id = project.project_id()))]
+    pub async fn store_project(&self, project: Project) -> Result<Project> {
+        if let Ok(project_identity) = project.project_identity() {
+            self.identities_verification
+                .update_identity_ignore_older(project_identity)
+                .await?;
+        }
+
+        if let Ok(authority_identity) = project.authority_identity() {
+            self.identities_verification
+                .update_identity_ignore_older(authority_identity)
+                .await?;
+        }
```

Comment:
Could we add a test for this?

Reply:
@adrianbenavides oh, nice that you caught it. I actually created a test for this but I didn't commit it accidentally. Thank you